### PR TITLE
Android Additional Tests

### PR DESCRIPTION
**Pull request by yschimke**
Add a smattering of additional unit tests to run under android.

image

### CODE REVIEW

File: okhttp-dnsoverhttps/src/test/java/okhttp3/dnsoverhttps/DnsOverHttpsTest.java [hunk 1]
```diff
@@ -36,13 +37,15 @@
...
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 import static java.util.Arrays.asList;
 import static java.util.Collections.singletonList;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.fail;
 
+@ExtendWith(MockWebServerExtension.class)
```
Comment (yschimke) [inline okhttp-dnsoverhttps/src/test/java/okhttp3/dnsoverhttps/DnsOverHttpsTest.java:48]: Need to register this manually in order for the junit5 android runner to pick it up.

File: okhttp-logging-interceptor/src/test/java/okhttp3/logging/LoggingEventListenerTest.java [hunk 2]
```diff
@@ -98,7 +101,7 @@ public void get() throws Exception {
         .assertLogMatch(
             "responseHeadersEnd: Response\\{protocol=http/1\\.1, code=200, message=OK, url="
                 + url
-                + "}")
+                + "\\}")
```
Comment (yschimke) [inline okhttp-logging-interceptor/src/test/java/okhttp3/logging/LoggingEventListenerTest.java:104]: Android regex is not as lax as JVM

**Comment by yschimke** [comment]
Results

```
> Task :android-test:connectedDebugAndroidTest
Starting 99 tests on Pixel_3a_API_30_x86(AVD) - 11

okhttp.android.test.OkHttpTest > testLoggingLevels[Pixel_3a_API_30_x86(AVD) - 11] SKIPPED

okhttp.android.test.OkHttpTest > testHttpRequestBlocked[Pixel_3a_API_30_x86(AVD) - 11] SKIPPED

okhttp.android.test.OkHttpTest > testSSLFeatures[Pixel_3a_API_30_x86(AVD) - 11] SKIPPED

okhttp.android.test.OkHttpTest > testBouncyCastleRequest[Pixel_3a_API_30_x86(AVD) - 11] SKIPPED

okhttp.android.test.OkHttpTest > testHttpRequestNotBlockedOnLegacyAndroid[Pixel_3a_API_30_x86(AVD) - 11] SKIPPED
        org.opentest4j.TestAbortedException: Assumption failed: assumption is not true
        at org.junit.jupiter.api.Assumptions.throwTestAbortedException(Assumptions.java:256)

okhttp3.brotli.BrotliInterceptorJavaApiTest > testApi[Pixel_3a_API_30_x86(AVD) - 11] SKIPPED
Starting 31 tests on 2.7_QVGA_API_22(AVD) - 5.1.1

okhttp.android.test.OkHttpTest > testBouncyCastleRequest[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testCertificatePinningFailure[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testCertificatePinningSuccess[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testConscryptRequest[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testCustomTrustManager[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testCustomTrustManagerWithAndroidCheck[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testDnsOverHttps[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testEventListener[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testHttpRequestBlocked[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testHttpRequestNotBlockedOnLegacyAndroid[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testLoggingLevels[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testMockWebserverRequest[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testPlatform[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testRequest[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testRequestUsesAndroidConscrypt[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testRequestUsesPlayProvider[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testRequestWithSniRequirement[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testSSLFeatures[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testSessionReuse[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.OkHttpTest > testUnderscoreRequest[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp.android.test.letsencrypt.LetsEncryptClientTest > get[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.brotli.BrotliInterceptorJavaApiTest > testApi[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.brotli.BrotliInterceptorTest > testFailsUncompress[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.brotli.BrotliInterceptorTest > testNoUncompress[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.brotli.BrotliInterceptorTest > testSkipUncompressNoContentResponse[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.brotli.BrotliInterceptorTest > testUncompressBrotli[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.brotli.BrotliInterceptorTest > testUncompressGzip[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.dnsoverhttps.DnsOverHttpsTest > failOnBadResponse[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.dnsoverhttps.DnsOverHttpsTest > failOnExcessiveResponse[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.dnsoverhttps.DnsOverHttpsTest > failure[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.dnsoverhttps.DnsOverHttpsTest > getIpv6[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.dnsoverhttps.DnsOverHttpsTest > getOne[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.dnsoverhttps.DnsOverHttpsTest > usesCache[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.dnsoverhttps.DnsOverHttpsTest > usesCacheOnlyIfFresh[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.dnsoverhttps.DnsRecordCodecTest > testGoogleDotComDecodingFromGoogleIPv6[2.7_QVGA_API_22(AVD) - 5.1.1] FAILED
        java.lang.NoSuchMethodError: No interface method stream()Ljava/util/stream/Stream; in class Ljava/util/Collection; or its super classes (declaration of 'java.util.Collection' appears in /system/framework/core-libart.jar)
        at org.assertj.core.util.Streams.stream(Streams.java:32)

okhttp3.dnsoverhttps.DnsRecordCodecTest > testGoogleDotComDecodingFromCloudflare[2.7_QVGA_API_22(AVD) - 5.1.1] FAILED
        java.lang.NoSuchMethodError: No interface method stream()Ljava/util/stream/Stream; in class Ljava/util/Collection; or its super classes (declaration of 'java.util.Collection' appears in /system/framework/core-libart.jar)
        at org.assertj.core.util.Streams.stream(Streams.java:32)

okhttp3.dnsoverhttps.DnsRecordCodecTest > testGoogleDotComDecodingFromGoogle[2.7_QVGA_API_22(AVD) - 5.1.1] FAILED
        java.lang.NoSuchMethodError: No interface method stream()Ljava/util/stream/Stream; in class Ljava/util/Collection; or its super classes (declaration of 'java.util.Collection' appears in /system/framework/core-libart.jar)
        at org.assertj.core.util.Streams.stream(Streams.java:32)

okhttp3.logging.HttpLoggingInterceptorTest > basicChunkedResponseBody[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > basicGet[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED
> IDLE
okhttp3.logging.HttpLoggingInterceptorTest > basicPost[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > basicResponseBody[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > bodyGet[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > bodyGet204[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > bodyGet205[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > bodyGetMalformedCharset[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > bodyPost[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > bodyResponseBody[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > bodyResponseBodyChunked[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > bodyResponseGzipEncoded[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > bodyResponseUnknownEncoded[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > connectFail[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > duplexRequestsAreNotLogged[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > headersAreRedacted[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > headersGet[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > headersPost[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > headersPostNoContentType[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > headersPostNoLength[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > headersPostWithHeaderOverrides[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > headersResponseBody[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > http2[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > levelGetter[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > none[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > oneShotRequestsAreNotLogged[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > responseBodyIsBinary[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > setLevelShouldPreventNullValue[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.HttpLoggingInterceptorTest > setLevelShouldReturnSameInstanceOfInterceptor[2.7_QVGA_API_22(AVD) - 5.1.1]
 SKIPPED

okhttp3.logging.IsProbablyUtf8Test > isProbablyUtf8[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.LoggingEventListenerTest > connectFail[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.LoggingEventListenerTest > dnsFail[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.LoggingEventListenerTest > get[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.LoggingEventListenerTest > post[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.LoggingEventListenerTest > secureGet[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.logging.LoggingEventListenerTest > testCacheEvents[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.sse.internal.EventSourceHttpTest > badContentType[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.sse.internal.EventSourceHttpTest > badResponseCode[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.sse.internal.EventSourceHttpTest > event[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.sse.internal.EventSourceHttpTest > fullCallTimeoutAppliesToSetup[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.sse.internal.EventSourceHttpTest > fullCallTimeoutDoesNotApplyOnceConnected[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.sse.internal.EventSourceHttpTest > retainsAccept[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.sse.internal.EventSourceHttpTest > setsMissingAccept[2.7_QVGA_API_22(AVD) - 5.1.1] SKIPPED

okhttp3.dnsoverhttps.DnsOverHttpsTest > usesCache[Pixel_3a_API_30_x86(AVD) - 11] FAILED
        java.net.UnknownHostException: google.com
        at okhttp3.dnsoverhttps.DnsOverHttps.throwBestFailure(DnsOverHttps.kt:176)

okhttp3.logging.HttpLoggingInterceptorTest > duplexRequestsAreNotLogged[Pixel_3a_API_30_x86(AVD) - 11] SKIPPED
        org.opentest4j.TestAbortedException: Assumption failed: assumption is not true
        at org.junit.jupiter.api.Assumptions.throwTestAbortedException(Assumptions.java:256)

okhttp3.logging.LoggingEventListenerTest > secureGet[Pixel_3a_API_30_x86(AVD) - 11] SKIPPED
        org.opentest4j.TestAbortedException: Assumption failed: assumption is not true
        at org.junit.jupiter.api.Assumptions.throwTestAbortedException(Assumptions.java:256)
```

**Comment by yschimke** [comment]
image

**Comment by yschimke** [comment]
Ignoring these test only failures for now

```
okhttp3.dnsoverhttps.DnsRecordCodecTest > testGoogleDotComDecodingFromGoogleIPv6[2.7_QVGA_API_22(AVD) - 5.1.1] FAILED
        java.lang.NoSuchMethodError: No interface method stream()Ljava/util/stream/Stream; in class Ljava/util/Collection; or its super classes (declaration of 'java.util.Collection' appears in /system/framework/core-libart.jar)
        at org.assertj.core.util.Streams.stream(Streams.java:32)

okhttp3.dnsoverhttps.DnsRecordCodecTest > testGoogleDotComDecodingFromCloudflare[2.7_QVGA_API_22(AVD) - 5.1.1] FAILED
        java.lang.NoSuchMethodError: No interface method stream()Ljava/util/stream/Stream; in class Ljava/util/Collection; or its super classes (declaration of 'java.util.Collection' appears in /system/framework/core-libart.jar)
        at org.assertj.core.util.Streams.stream(Streams.java:32)

okhttp3.dnsoverhttps.DnsRecordCodecTest > testGoogleDotComDecodingFromGoogle[2.7_QVGA_API_22(AVD) - 5.1.1] FAILED
        java.lang.NoSuchMethodError: No interface method stream()Ljava/util/stream/Stream; in class Ljava/util/Collection; or its super classes (declaration of 'java.util.Collection' appears in /system/framework/core-libart.jar)
        at org.assertj.core.util.Streams.stream(Streams.java:32)
```

File: okhttp-dnsoverhttps/src/test/java/okhttp3/dnsoverhttps/DnsOverHttpsTest.java (hunk 1 above)
Comment (swankjesse) [reply okhttp-dnsoverhttps/src/test/java/okhttp3/dnsoverhttps/DnsOverHttpsTest.java:48]: Can we set the system property to turn on ‘automatic’ extensions?

File: okhttp-logging-interceptor/src/test/java/okhttp3/logging/LoggingEventListenerTest.java (hunk 2 above)
Comment (swankjesse) [reply okhttp-logging-interceptor/src/test/java/okhttp3/logging/LoggingEventListenerTest.java:104]: Nice.

**Review by swankjesse — APPROVED** [review]

File: okhttp-dnsoverhttps/src/test/java/okhttp3/dnsoverhttps/DnsOverHttpsTest.java (hunk 1 above)
Comment (yschimke) [reply okhttp-dnsoverhttps/src/test/java/okhttp3/dnsoverhttps/DnsOverHttpsTest.java:48]: Forgot about that

https://github.com/mannodermaus/android-junit5/issues/229

Also raised a question of O minimum, whch seems debateable.  https://github.com/mannodermaus/android-junit5/issues/228

Comment (yschimke) [reply okhttp-dnsoverhttps/src/test/java/okhttp3/dnsoverhttps/DnsOverHttpsTest.java:48]: For follow up

**Pull request merged**